### PR TITLE
Ch7 Ch8 Ch10 generic showPhoneType

### PR DESCRIPTION
**Pull request by milesfrain**
Since we cover deriving from generic in Ch6, we might as well use in it later chapters.

### CODE REVIEW

File: exercises/chapter8/src/Data/AddressBook.purs [hunk 1]
```diff
@@ -17,11 +19,10 @@ data PhoneType
...
   | OtherPhone
 
+derive instance genericPhoneType :: Generic PhoneType _
+
 instance showPhoneType :: Show PhoneType where
-  show HomePhone  = "HomePhone"
-  show WorkPhone  = "WorkPhone"
-  show CellPhone  = "CellPhone"
-  show OtherPhone = "OtherPhone"
+  show = genericShow
```
Comment (milesfrain) [inline exercises/chapter8/src/Data/AddressBook.purs:26]: Same total number of lines, but the generic version is a lot more maintainable. Hopefully this doesn't add too much noise / confusion / complexity for readers.

File: exercises/chapter7/src/Data/AddressBook.purs [hunk 2]
```diff
@@ -21,18 +23,26 @@ data PhoneType
...
+-- Manually-written Show instance
 instance showPhoneType :: Show PhoneType where
-  show HomePhone = "HomePhone"
-  show WorkPhone = "WorkPhone"
-  show CellPhone = "CellPhone"
+  show HomePhone  = "HomePhone"
+  show WorkPhone  = "WorkPhone"
+  show CellPhone  = "CellPhone"
   show OtherPhone = "OtherPhone"
+-}
```
Comment (milesfrain) [inline exercises/chapter7/src/Data/AddressBook.purs:45]: Keeping manual version for reference only in Ch7

File: exercises/chapter10/src/Data/AddressBook.purs [hunk 3]
```diff
@@ -26,11 +26,12 @@ data PhoneType
   | OtherPhone
 
 -- ANCHOR: PhoneType_generic
-derive instance genericPhoneType    :: Generic    PhoneType _
-instance        encodeJsonPhoneType :: EncodeJson PhoneType where encodeJson = genericEncodeJson
-instance        decodeJsonPhoneType :: DecodeJson PhoneType where decodeJson = genericDecodeJson
+derive instance genericPhoneType :: Generic PhoneType _
```
Comment (milesfrain) [inline exercises/chapter10/src/Data/AddressBook.purs:29]: I think it's clearer to keep alignment of the derived instance separate from the other instance definitions.

**Comment by milesfrain** [comment]
@razcore-art This PR is related to what you've been working on. Checking if you have any feedback before I merge.

**Comment by razcore-art** [comment]
Yep, looks all good to me

**Pull request merged**